jax/_src/lax/control_flow/loops.py

```diff
@@ -2157,8 +2157,46 @@ def fori_loop(lower, upper, body_fun, init_val):
 
 ### map and miscellaneous rules
 
+def _transpose_with_axes(x, axes, *, move_front: bool):
+  flat_axes: list[int] = broadcast_prefix(axes, x, is_leaf=lambda x: x is None)
+  leaves, treedef = tree_flatten(x)
+
+  if move_front:
+    move_axes = lambda leaf, axis: jax.numpy.moveaxis(leaf, axis, 0)
+  else:
+    move_axes = lambda leaf, axis: jax.numpy.moveaxis(leaf, 0, axis)
+
+  leaves = [
+    move_axes(leaf, axis)
+    for leaf, axis in zip(leaves, flat_axes)
+  ]
+  return treedef.unflatten(leaves)
+
+def _batch_and_remainder(x, batch_size: int):
+  leaves, treedef = tree_flatten(x)
+
+  scan_leaves = []
+  remainder_leaves = []
+
+  for leaf in leaves:
+    num_batches, remainder = divmod(leaf.shape[0], batch_size)
+    total_batch_elems = num_batches * batch_size
+    scan_leaves.append(leaf[:total_batch_elems].reshape((num_batches, batch_size) + leaf.shape[1:]))
+    remainder_leaves.append(leaf[total_batch_elems:])
+
+  scan_tree = treedef.unflatten(scan_leaves)
+  remainder_tree = treedef.unflatten(remainder_leaves)
+  return scan_tree, remainder_tree
+
 @api_boundary
-def map(f, xs):
+def map(
+  f,
+  xs,
+  *,
+  in_axes: int | None | Any = 0,
+  out_axes: Any = 0,
```

Comment:
Can we not add in_axes/out_axes in this PR? They're complicated (e.g. it's not clear that we should support `out_axes=None`), implemented with a transpose that the calling code could do (rather than slicing along the given axes), and the test coverage here isn't good. Let's just make this PR about `batch_size`.